### ISSUE TITLE
Message filtering

### ISSUE THREAD

**Issue by Chumper** [bennidi/mbassador#90]
I read about the message filter, but the wiki just lists some headers. 
How about continuing the work there?

**Comment by bennidi** [comment]
Hi, sorry for not getting back to you. It seems I somehow forgot about this issue. Are you still interested? I have the intuition the your requirement can be implemented quite easily with the functionality that is already there. If you are interested you can lay out your use case in more detail and I will see what I can do.

**Comment by Chumper** [comment]
Yes, even so i now use Spring Reactor i will lay out a use case for you so you can decide if th euse case is common or not :)

**Comment by message** [comment]
@Chumper hope you're home and safe

**Comment by Chumper** [comment]
Yes, took me quite a while to get home but i will try to present you my current use case:

Currently i am writing an app with libgdx, it is a puzzle game.
In the game i use the messages to send events of actions i took in the code, e.g. `PuzzleCreated`, `PuzzleSolved`, `PuzzleRated` and so on.
In most cases this is working perfectly but in specific cases i need to react on the action.

For example after a puzzle is rated i want to transmit the rating to the server that can happen without waiting for a response.

Another use case is puzzle generation: A user can generate a puzzle on the device. After the user presses `Generate` i send an `GeneratePuzzleIntent` and will show a loading indicator on the screen. At this point i am waiting for a `PuzzleGenerated` event to show the generated puzzle on the screen.

In order for this to work, i need to listen on dynamic topics or wait until a message returns to the sender which is not possible at the moment.

Therefore i would appreciate a consumer i can pass to the bus which is executed after all consumers handled the message or the possibility to create a second dynamic topic where all handlers can respond to.

I used `reactor` http://projectreactor.io/docs/reference/#bus-request-reply in another project and i needed the request response pattern 1-2 times. 
It would be great if MBassador could offer something similar even with some kind of more code effort.
